tests/resource_tests/calculator_tests/test_ubr_calculator.py

```diff
@@ -0,0 +1,322 @@
+                         self.ubr.ubr_sexual_gender_minority(UBRValueEnum.RBR, UBRValueEnum.UBR))
+        self.assertEqual(UBRValueEnum.UBR,
+                         self.ubr.ubr_sexual_gender_minority(UBRValueEnum.NullSkip, UBRValueEnum.UBR))
+        self.assertEqual(1,
+                         self.ubr.ubr_sexual_gender_minority(UBRValueEnum.RBR, UBRValueEnum.UBR))
+
+        # Test with Null/Skip Values. int(NullSkip) == 2.
+        self.assertEqual(UBRValueEnum.NullSkip,
+                         self.ubr.ubr_sexual_gender_minority(UBRValueEnum.NullSkip, UBRValueEnum.NullSkip))
+        self.assertEqual(2,
+                         self.ubr.ubr_sexual_gender_minority(UBRValueEnum.NullSkip, UBRValueEnum.NullSkip))
+
+    def test_ubr_ethnicity(self):
+        """
+        UBR Calculator Test - Ethnicity
+        Note: Multiple value UBR calculation.
+        """
+        # Test with Null and PMI_Skip values
+        # Note: Multiple choice question, pass comma-delimited string.
+        self.assertEqual(self.ubr.ubr_ethnicity(None), UBRValueEnum.NullSkip)
+        self.assertEqual(self.ubr.ubr_ethnicity('PMI_Skip'), UBRValueEnum.NullSkip)
+
+        # Test UBR value
+        self.assertEqual(self.ubr.ubr_ethnicity('WhatRaceEthnicity_AIAN'), UBRValueEnum.UBR)
+        self.assertEqual(self.ubr.ubr_ethnicity('WhatRaceEthnicity_Black'), UBRValueEnum.UBR)
+        self.assertEqual(self.ubr.ubr_ethnicity('WhatRaceEthnicity_MENA'), UBRValueEnum.UBR)
+        self.assertEqual(self.ubr.ubr_ethnicity('WhatRaceEthnicity_Hispanic'), UBRValueEnum.UBR)
+        self.assertEqual(self.ubr.ubr_ethnicity('WhatRaceEthnicity_NHPI'), UBRValueEnum.UBR)
+        self.assertEqual(self.ubr.ubr_ethnicity('WhatRaceEthnicity_RaceEthnicityNoneOfThese'), UBRValueEnum.UBR)
+        self.assertEqual(self.ubr.ubr_ethnicity('WhatRaceEthnicity_White,WhatRaceEthnicity_NHPI'), UBRValueEnum.UBR)
+
+        # Test RBR value
+        self.assertEqual(self.ubr.ubr_ethnicity('WhatRaceEthnicity_White'), UBRValueEnum.RBR)
+        self.assertEqual(self.ubr.ubr_ethnicity('PMI_PreferNotToAnswer'), UBRValueEnum.RBR)
+
+        # Bad or unknown value will default to UBR
+        self.assertEqual(self.ubr.ubr_ethnicity('ABC-123'), UBRValueEnum.UBR)
+
+
+    def test_ubr_geography(self):
+        """
+        UBR Calculator Test - Geography
+        """
+        consent_2014 = datetime(2019, 7, 1).date()  # Hit 2014 version of rural zipcodes.
+        consent_2020 = RURAL_2020_CUTOFF  # Hit 2020 version of rural zipcodes.
+
+        zip_2014 = '35052'  # Only in 2014 file.
+        zip_2020 = '35063'  # Only in 2020 file.
+        zip_both = '72529'  # In both the 2014 file and 2020 file.
+        zip_none = '58493'  # Zip code not in either file.
+
+        # Test with Null and PMI_Skip values
+        self.assertEqual(self.ubr.ubr_geography(consent_2014, None), UBRValueEnum.NullSkip)
+        self.assertEqual(self.ubr.ubr_geography(consent_2020, None), UBRValueEnum.NullSkip)
```

Comment:
Is PMI_Skip a code that needs to be tested as the 2nd param (per comment)?

Reply:
Yes, I just copied that comment. There are no PMI_Skip zip code values in the data, but there are Nulls.